File: base-class-module/src/main/java/jingya/com/base_class_module/BaseAdapter/BaseListViewAdapter.java
package jingya.com.base_class_module.BaseAdapter;

import android.content.Context;
import android.databinding.DataBindingUtil;
import android.databinding.ViewDataBinding;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;

import java.util.List;

/**
 * @author Kuky
 */
public abstract class BaseListViewAdapter<T, VB extends ViewDataBinding> extends BaseAdapter {
    protected VB mViewBinding;
    protected List<T> mData;
    protected Context mContext;
    protected int selectedPosition = -1;
    protected LayoutInflater mInflater;

    public void updateAdapterData(List<T> data) {
        this.mData = data;
        notifyDataSetChanged();
    }

    public List<T> getAdapterData() {
        return mData;
    }

    public void setSelectedItemPosition(int position) {
        this.selectedPosition = position;
    }

    public BaseListViewAdapter(Context context) {
        this.mContext = context;
        this.mInflater = LayoutInflater.from(mContext);
    }

    @Override
    public int getCount() {
        return mData == null ? 0 : mData.size();
    }

    @Override
    public Object getItem(int position) {
        return mData.get(position);
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        if (convertView == null) {
            mViewBinding = DataBindingUtil.inflate(mInflater, getLayoutId(), parent, false);
        } else {
            mViewBinding = DataBindingUtil.getBinding(convertView);
        }
        setVariable(mViewBinding, mData.get(position));
        return mViewBinding.getRoot();
    }

    protected abstract void setVariable(VB viewBinding, T t);

    protected abstract int getLayoutId();
}

File: base-kt-module/src/main/java/com/kuky/base_kt_module/BaseAdapters/BaseListViewAdapter.kt
package com.kuky.baselib.baseAdapter

import android.content.Context
import android.databinding.DataBindingUtil
import android.databinding.ViewDataBinding
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter

/**
 * @author Kuky
 */
abstract class BaseListViewAdapter<T : Any, VB : ViewDataBinding>(context: Context, data: MutableList<T>?) : BaseAdapter() {
    protected var mContext: Context = context
    protected var mData: MutableList<T>? = data
    protected lateinit var mViewBinding: VB
    protected val mInflater: LayoutInflater
    protected var mSelectedPosition: Int = -1

    init {
        this.mInflater = LayoutInflater.from(mContext)
    }

    fun updateAdapterData(data: MutableList<T>) {
        this.mData = data
        notifyDataSetChanged()
    }

    fun getAdapterData(): MutableList<T>? {
        return mData
    }

    fun setSelectedItemPosition(position: Int) {
        this.mSelectedPosition = position
        notifyDataSetChanged()
    }

    override fun getCount(): Int {
        return if (mData == null) 0 else mData!!.size
    }

    override fun getItem(position: Int): Any {
        return mData!![position]
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup?): View {
        mViewBinding =
                if (convertView == null) {
                    DataBindingUtil.inflate(mInflater, getItemLayoutId(), parent, false)
                } else {
                    DataBindingUtil.getBinding(convertView)!!
                }
        setVariable(mViewBinding, position)
        return mViewBinding.root
    }

    abstract fun setVariable(viewBinding: VB, position: Int)

    abstract fun getItemLayoutId(): Int
}